components/raftstore-v2/src/raft/peer.rs

```diff
@@ -133,7 +135,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         let raft_group = RawNode::new(&raft_cfg, storage, &logger)?;
         let region = raft_group.store().region_state().get_region().clone();
 
-        let flush_state: Arc<FlushState> = Arc::default();
+        let flush_state: Arc<FlushState> = Arc::new(FlushState::new(applied_index));
```

Comment:
It appears this change will lead to a panic. 
"
[2022/12/28 13:22:52.755 -08:00] [FATAL] [lib.rs:496] ["to_commit 5 is out of range [last_index 0], raft_id: 213, peer_id: 213, region_id: 210"] [backtrace="   0: tikv_util::set_panic_hook::{{closure}}\n   1: std::panicking::rust_panic_with_hook\n   2: std::panicking::begin_panic_handler::{{closure}}\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\n   4: rust_begin_unwind\n   5: core::panicking::panic_fmt\n   6: raft::raft_log::RaftLog<T>::commit_to\n   7: raft::raft::Raft<T>::restore\n   8: raft::raft::Raft<T>::handle_snapshot\n   9: raft::raft::Raft<T>::step_follower\n  10: raft::raft::Raft<T>::step\n  11: raft::raw_node::RawNode<T>::step\n  12: raftstore_v2::operation::ready::<impl raftstore_v2::raft::peer::Peer<EK,ER>>::on_raft_message\n  13: raftstore_v2::fsm::peer::PeerFsmDelegate<EK,ER,T>::on_msgs\n  14: <raftstore_v2::batch::store::StorePoller<EK,ER,T> as batch_system::batch::PollHandler<raftstore_v2::fsm::peer::PeerFsm<EK,ER>,raftstore_v2::fsm::store::StoreFsm>>::handle_normal\n  15: batch_system::batch::Poller<N,C,Handler>::poll\n  16: std::sys_common::backtrace::__rust_begin_short_backtrace\n  17: core::ops::function::FnOnce::call_once{{vtable.shim}}\n  18: std::sys::unix::thread::Thread::new::thread_start\n  19: start_thread\n             at /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477:8\n  20: clone\n             at /build/glibc-SzIz7B/glibc-2.31/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95\n"] [location=/home/qi/.cargo/git/checkouts/raft-rs-42b8049ef2e3af07/2357cb2/src/raft_log.rs:292] [thread_name=rs-1-1]"